devlib/utils/gem5.py

```diff
@@ -39,5 +46,6 @@ def iter_statistics_dump(stats_file):
             if res:
                 k = res.group("key")
                 v = res.group("value").split()
+                v = map(best_effort_numeric, v)
```

Comment:
I think it makes sense to raise an exception here. We care about the format of the stats file in this function only so if there is a formatting issue it should be raised here. I can replace line 49 with `v = map(numeric, v)` and remove `best_effort_numeric`. Is that OK for you ? If you think it's necessary I can also `try ... catch` the cast to raise a more readable exception.
And, for safety, we should probably wait for  #162 to be merged before merging this one.